Week-3/Exercise 3_1.js

```diff
@@ -0,0 +1,39 @@
+//memorize methods for add function 
+function memoize(fn)
+{
+    const log ={};
+    return function(...args)
+    {
+        const key = args.toString();
+        if(!log[key])
+        {
+            log[key] = fn(...args);
+        }
+        return log[key]; 
+    };
+
+}
+
+// function add, adds two numbers
+function add(a,b)
+{
+    return (a+b);
+    
+}
+
+//Reference to memorize add function
+const test1= memoize(add);
+
+function timeCalculate(fn)
+{
+    console.time();
+    fn();
+    console.timeEnd();
+}
+ 
+// call add function to add number and note time required to add 
+timeCalculate(() => console.log(test1(5,2)));
```

Comment:
will it return from cache if i change the position of paramters? like first time i call test1(5,2) which does the calculation , second time when i call test1(2,5) it should return from cache.. check this usecase if not please handle this